geniza/common/tests.py

```diff
@@ -329,3 +331,25 @@ def test_item_count(self):
         item_count = tag_admin.item_count(qs.first())
 
         assert item_count == 2
+
+
+def test_echo():
+    echo = Echo()
+
+    value = random.random()
+    assert value is echo.write(value)
+
+    with Echo() as e:
+        assert type(e) == Echo
+
+
+@pytest.mark.django_db
+def test_base_exporter():
+    exporter = Exporter()
+
+    # raises correct error?
+    try:
+        exporter.get_export_data_dict(obj=None)
+        assert False  #
+    except NotImplementedError:
+        pass
```

Comment:
oh, actually — you can use pytest raises for this

https://docs.pytest.org/en/7.1.x/how-to/assert.html#assertions-about-expected-exceptions

should look something like this:

```suggestion
def test_base_exporter():
    exporter = Exporter()

    # raises correct error?
    with pytest.raises(NotImplementedError):
        exporter.get_export_data_dict(obj=None)
```

Pretty sure this doesn't hit the db, so you don't need the django_db mark